### PR TITLE
Tooltips, style changes, add tippy.js, replace .action class with relevant naming

### PR DESCRIPTION
**Pull request by calvinmorett**
### Tooltips Added with Tippy.js, CSS class changes, Style Changes, Off Screen Nav Toggle

Add offscreen nav toggle to clear screen space for note editing focus
https://drive.google.com/file/d/1xCo18UAvIqXJwzJNcvetqAJ6zHnT7dsT

- added to public/js/: tippy.js, tippy_settings.js, popper.min.js(tippys dependency)
- `(app.blade.php)` added srcs to tippy's .js/.css, custom settings.js, and popper.js
- `(nav.css)` smoother transitions added on hovers on notes and buttons
- `(nav.css)` added better spacing for notes themselves
- `(editor.css)` removed` <br> `overuse, added margin-top to bottom-links for spacing instead
- `(editor.css)` added more spacing between note actions, delete, star, perm-del, restore
- `(nav.css)` added bigger buttons to create notebooks and notes
- `(nav.css)` muted notebooks/notes colors
- `(nav.css)` muted spacing between notes to a more subtle bottom border
- `(nav.css)` edit .action as button to .newnotebok and .newnote respectively to their corresponding buttons
- `(nav.css + app.blade.php)` editted css class values for .action to be replaced with .newnotebook and .newnote respectively
- `(nav.css)` Styled text bigger if inside of an empty list

_Next Priority or Issues_
- Responsive design
- `(nav.css)` .notelist:hover added when a lot of notes are inside a notebook to scroll
- Tinymce might be removed, so it doesn't make sense to style anything in the editor view for now, but I think most people would prefer a bigger text box to start.

### Bugs
- If you hover an active note Star / Del, then go to the Trash and hover Restore / Perma Del, the tooltip reads Star / Del over those items until refreshed.
image
- When right clicking a notebook it sets the context menu far to the right (Suggested revision: revert/deny current implementation of sidebar toggle)
image
- Default note titles appear still
image
- Starred Notes should have a warning that they will be removed from being Starred if they are deleted, Starred notes that are deleted in the Trashed Notes still appear Starred.
image

### CODE REVIEW

File: resources/views/app.blade.php [hunk 1]
```diff
@@ -130,44 +132,26 @@ class="action center"
 			</div>
 
 		</div>
+		</div><!-- end nav wrapper - for off screen nav toggle -->
 
 
 		{{-- Note Content Editor --}}
+<input type="checkbox" id="nav-trigger" class="nav-trigger" />
```
Comment (calvinmorett) [inline resources/views/app.blade.php]: Toolbar Toggle Addition

File: resources/views/app.blade.php [hunk 2]
```diff
@@ -130,44 +132,26 @@ class="action center"
...
+<label for="nav-trigger"></label>
+<div class="site-wrap">
```
Comment (calvinmorett) [inline resources/views/app.blade.php]: Toolbar Toggle Addition

File: resources/views/app.blade.php [hunk 3]
```diff
@@ -130,44 +132,26 @@ class="action center"
 			</div>
 
 		</div>
+		</div><!-- end nav wrapper - for off screen nav toggle -->
 
 
 		{{-- Note Content Editor --}}
+<input type="checkbox" id="nav-trigger" class="nav-trigger" />
+<label for="nav-trigger"></label>
```
Comment (calvinmorett) [inline resources/views/app.blade.php]: Toolbar Toggle Addition

File: resources/views/app.blade.php [hunk 4]
```diff
@@ -91,26 +100,19 @@ class="notebook_link"
...
 			</div>
-
 		</nav>
 
 
 		{{-- List Notes --}}
 		<div class="notes-list">
 			<button
-				class="action center"
+				class="newnote center"
```
Comment (calvinmorett) [inline resources/views/app.blade.php]: .action is too common of a class name and was causing issues with other css, I think Tippy.css

File: resources/views/app.blade.php [hunk 5]
```diff
@@ -53,12 +61,13 @@
...
 
 		{{-- Notebooks Sidebar --}}
+		<div class="nav-wrap">
 		<nav>
 			<div class="logo-wrapper">
 				<a class="logo ondark is-small" href="/">Dopenote</a>
 			</div>
 
-			<button class="action center" v-on:click="create_notebook()" :disabled="waiting_for_ajax">New Notebook</button>
+			<button class="newnotebook center" v-on:click="create_notebook()" :disabled="waiting_for_ajax">New Notebook</button>
```
Comment (calvinmorett) [inline resources/views/app.blade.php]: .action is too common of a class name and was causing issues with other css, I think Tippy.css

File: resources/views/app.blade.php [hunk 6]
```diff
@@ -31,11 +32,18 @@
...
+	
+	{{-- tooltips --}}
+	<script src="https://unpkg.com/popper.js"></script> <!-- tippy uses popper as a depedency -->
+	<script src="https://unpkg.com/tippy.js"></script>
+	
 	{{-- Custom scripts --}}
 	<script src="/js/editor.js"></script>
 	<script src="/js/tinymce_init.js"></script>
 	<script src="/js/vue_init.js"></script>
+	<script src="/js/popper.min.js"></script>
```
Comment (calvinmorett) [inline resources/views/app.blade.php]: Should move to CDN these, but I was unable to get the tooltip to work without downloading them and placing them inside js root

File: resources/views/app.blade.php [hunk 7]
```diff
@@ -31,11 +32,18 @@
...
+	<script src="/js/tippy.js"></script>
+	<script src="/js/tippy_settings.js"></script>
```
Comment (calvinmorett) [inline resources/views/app.blade.php]: Quick settings js for tippy, perhaps we can merge this with another js so we don't have to load so many custom js files

File: public/js/tippy_settings.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+new Tippy('.note-action',{
```
Comment (calvinmorett) [inline public/js/tippy_settings.js]: original was targeting an id of Tooltip but if we can target classes, and note-actions was what you wanted to be targeted for a tooltip we can just use these.

File: public/css/nav.css [hunk 9]
```diff
@@ -1,11 +1,75 @@
+.nav-wrap{
```
Comment (calvinmorett) [inline public/css/nav.css]: Everything added, will be revised when I start working on Responsive.

File: resources/views/app.blade.php [hunk 10]
```diff
@@ -31,11 +32,18 @@
 	{{-- Rich-text editor --}}
 	<script src="https://cdnjs.cloudflare.com/ajax/libs/tinymce/5.0.0/tinymce.min.js"></script>
 	<script src="https://cdnjs.cloudflare.com/ajax/libs/tinymce/5.0.0/plugins/textpattern/plugin.min.js"></script>
-
+	
+	{{-- tooltips --}}
```
Comment (xy2z) [inline resources/views/app.blade.php]: You added these twice?

Comment (xy2z) [reply resources/views/app.blade.php]: Popper.js and tippy.js is added via https://unpkg.com/ and "/js/popper.min.js"

File: resources/views/app.blade.php (hunk 7 above)
Comment (xy2z) [reply resources/views/app.blade.php]: Yes all JS and CSS should be minified to 1 file each, but that's a nice-to-have task for later 👍

File: resources/views/app.blade.php [hunk 11]
```diff
@@ -72,7 +81,7 @@
 				</a>
 			</div>
 
-			<br>
+			<br> <!-- do we need this? -->
```
Comment (xy2z) [inline resources/views/app.blade.php]: nope

File: resources/views/app.blade.php [hunk 12]
```diff
@@ -130,44 +132,26 @@ class="action center"
...
-						title="Star note"
-						></button>
-
-					<button
-						v-on:click="delete_note(getActiveNote())"
-						:disabled="waiting_for_ajax"
-						class="note-action fas fa-trash"
-						title="Trash note"
-						></button>
+					<button v-on:click="toggle_star(getActiveNote())":disabled="waiting_for_ajax" class="note-action tippy fas fa-star" v-bind:class="getStarClass(getActiveNote())" title="Star Note"></button>
```
Comment (xy2z) [inline resources/views/app.blade.php]: please add the linebreaks back, it's more readable

File: public/css/nav.css [hunk 13]
```diff
@@ -131,3 +197,80 @@ nav .nav-notebooks a.sortable-ghost {
 .nav-bottom-links a:hover {
 	color: #999;
 }
+nav {
```
Comment (xy2z) [inline public/css/nav.css]: You have 2 `nav` styles

File: resources/views/app.blade.php (hunk 1 above)
Comment (xy2z) [reply resources/views/app.blade.php]: Needs indentation

**Comment by xy2z** [comment]
Great! 👍
Just a couple notes 

Generally I'd like much smaller PR's, so it's 1 PR per issue/task. Eg. the tooltip is 1 task, style changes is 1 change, image optimization is another, etc.. Just so it's easier to review and comment, and there's not any bottleneck holding other fixes back, and it's clear what the PR is.

I'd appreciate it if you used only tabs as indentation instead of spaces, so it's consistent through all css and js files.

Add space after `:` so its `color: #fff` instead of `color:#fff`

The html in file `app.blade.php` seems to have incorrect indentation, with the new elements added.

Also I've added some comments

**Comment by calvinmorett** [comment]
Alright, I was also planning on running linter through the CSS, to keep it cleaner.

I'll make sure in the future to separate requests into easier to understand and accept/revise. My fault for adding so many things in one pass. Let me close this, and I will revise the changes you need and then resubmit separately.

**Comment by xy2z** [comment]
Thanks!

**Pull request closed without merging**